### ISSUE TITLE
Spans for macros as impl items seem to be wrong

### ISSUE THREAD

**Issue by nrc** [rust-lang/rust#30067]
e.g.,

```
impl Foo { add_fun!(); }

impl Blah {
    fn boop() {}
    add_fun!();
}
```

which gives something like:

```
ImplItem {
    ...
    node: Macro(Spanned { 
              node: Mac_ { path: path(add_fun), tts: [], ctxt: SyntaxContext(0) }, 
              span: Span { lo: BytePos(868), hi: BytePos(869), expn_id: ExpnId(4294967295) } 
    }), 
    span: Span { lo: BytePos(858), hi: BytePos(869), expn_id: ExpnId(4294967295) } 
}
```

Note that the span of the macro is different from the span on the ImplItem, and is just the span for the `;`.

**Comment by nrc** [comment]
cc @DanielJCampbell

**Comment by sanxiyn** [comment]
This was fixed by #30075.
